server/sonar-process-monitor/src/test/java/org/sonar/application/cluster/HazelcastClusterTest.java

```diff
@@ -0,0 +1,203 @@
+      assertThat(hzCluster.isOperational(ProcessId.COMPUTE_ENGINE)).isFalse();
+
+      // Connect via Hazelcast client to test values
+      HazelcastInstance hzInstance = createHazelcastClient(hzCluster);
+      ReplicatedMap<ClusterProcess, Boolean> operationalProcesses = hzInstance.getReplicatedMap(OPERATIONAL_PROCESSES);
+      assertThat(operationalProcesses)
+        .containsExactly(new AbstractMap.SimpleEntry<>(new ClusterProcess(hzCluster.getLocalUuid(), ProcessId.ELASTICSEARCH), Boolean.TRUE));
+    }
+  }
+
+  @Test
+  public void cluster_name_comes_from_configuration() {
+    TestAppSettings testAppSettings = newClusterSettings();
+    testAppSettings.set(CLUSTER_NAME, "a_cluster_");
+    ClusterProperties clusterProperties = new ClusterProperties(testAppSettings);
+    try (HazelcastCluster hzCluster = HazelcastCluster.builder(clusterProperties).build()) {
+      assertThat(hzCluster.getName()).isEqualTo("a_cluster_");
+    }
+  }
+
+  @Test
+  public void localUUID_must_not_be_empty() {
+    ClusterProperties clusterProperties = new ClusterProperties(newClusterSettings());
+    try (HazelcastCluster hzCluster = HazelcastCluster.builder(clusterProperties).build()) {
+      assertThat(hzCluster.getLocalUuid()).isNotEmpty();
+    }
+  }
+
+  @Test
+  public void when_a_process_is_set_operational_listener_must_be_triggered() {
+    ClusterProperties clusterProperties = new ClusterProperties(newClusterSettings());
+    try (HazelcastCluster hzCluster = HazelcastCluster.builder(clusterProperties).build()) {
+      AppStateListener listener = mock(AppStateListener.class);
+      hzCluster.addListener(listener);
+
+      // ElasticSearch is not operational
+      assertThat(hzCluster.isOperational(ProcessId.ELASTICSEARCH)).isFalse();
+
+      // Simulate a node that set ElasticSearch operational
+      HazelcastInstance hzInstance = createHazelcastClient(hzCluster);
+      ReplicatedMap<ClusterProcess, Boolean> operationalProcesses = hzInstance.getReplicatedMap(OPERATIONAL_PROCESSES);
+      operationalProcesses.put(new ClusterProcess(UUID.randomUUID().toString(), ProcessId.ELASTICSEARCH), Boolean.TRUE);
+      verify(listener, timeout(20_000)).onAppStateOperational(ProcessId.ELASTICSEARCH);
+      verifyNoMoreInteractions(listener);
+
+      // ElasticSearch is operational
+      assertThat(hzCluster.isOperational(ProcessId.ELASTICSEARCH)).isTrue();
+    }
+  }
+
+
+  @Test
+  public void registerSonarQubeVersion_publishes_version_on_first_call() {
+    ClusterProperties clusterProperties = new ClusterProperties(newClusterSettings());
+    try (HazelcastCluster hzCluster = HazelcastCluster.builder(clusterProperties).build()) {
+      hzCluster.registerSonarQubeVersion("1.0.0.0");
+
+      HazelcastInstance hzInstance = createHazelcastClient(hzCluster);
+      assertThat(hzInstance.getAtomicReference(SONARQUBE_VERSION).get())
+        .isNotNull()
+        .isInstanceOf(String.class)
+        .isEqualTo("1.0.0.0");
```

Comment:
no need for isNotNull nor isInstanceOf(String.class). They are both implicit with .isEqualTo("1.0.0.0")